runtime/src/launch/slurm-srun/launch-slurm-srun.c

```diff
@@ -57,6 +57,23 @@ typedef enum {
   unknown
 } sbatchVersion;
 
+// Find the default tmp directory. Try getting the tmp dir from the
+// ISO/IEC 9945 env var options first, then P_tmpdir, then "/tmp"
+static const char* getTmpDir() {
+  int i;
+  const char* possibleDirsInEnv[] = {"TMPDIR", "TMP", "TEMP", "TEMPDIR"};
+  for (i = 0; i < (sizeof(possibleDirsInEnv) / sizeof(char*)); i++) {
+    const char* curDir = getenv(possibleDirsInEnv[i]);
+    if (curDir != NULL) {
+      return curDir;
+    }
+  }
+#ifdef P_tmpdir
+  return P_tmpdir;
+#endif
```

Comment:
Didn't mean for that to be so long, I just got carried away with trying out the embedded code thing :)